template.yaml

```diff
@@ -1,68 +1,122 @@
+      eg, 2 means logs for last 2 days.
+    AllowedPattern: .+
+  CreateSecret:
+    AllowedValues:
+      - 'Yes'
+      - 'No'
+    Default: 'No'
+    Description: Yes -> Encrypt the Provided Environment variables HTTP_LOGS_ENDPOINT,
+      TOKEN using KMS and Secret Manager. No -> No encryption.
+    Type: String
   HttpLogsEndpoint:
     Type: String
     Description: Paste the URL for the HTTP Logs source.
-    AllowedPattern: ".+"
+    AllowedPattern: .+
     NoEcho: true
   Token:
     Type: String
     Description: Paste the TOKEN generated from Slack App.
-    AllowedPattern: ".+"
+    AllowedPattern: .+
     NoEcho: true
   EnableInfrequentChannels:
     Type: String
-    Description: "True -> Enable dividing channels into frequent and infrequent based on the last message time.
-                  False -> Send all public channels messages."
+    Description: True -> Enable dividing channels into frequent and infrequent based
+      on the last message time. False -> Send all public channels messages.
     AllowedValues:
       - true
       - false
     Default: false
-  InfrequentChannelsThresholdInHours:
-    Type: String
-    Description: Threshold in hours to make channels as infrequent based on last message time. For eg, 12 hours means if the message is not recived for 12 hours, channel will be marked as infrequent.
-    Default: 72
-  InfrequentChannelsMessagesFetchTimeInHours:
-    Type: String
-    Description: Time in hours to fetch messages for InFrequent channels. For eg, 12 hours means send infrequent channels messages every 12 hours.
-    Default: 12
   DatabaseName:
     Type: String
-    Description: "DynamoDB DataBase Table Name."
+    Description: DynamoDB Table Name. Use the table name to identify the Slack Workspace for which you are collecting logs. Do not use the same DataBase Name across multiple installations.
     Default: sumoslack
+
 Resources:
+  LambdaRole:
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: sts:AssumeRole
+            Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+        Version: '2012-10-17'
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+        - arn:aws:iam::aws:policy/AmazonDynamoDBFullAccess
+      Path: /
+    Type: AWS::IAM::Role
+
+  SecretPolicy:
+    Condition: create_secret
+    Properties:
+      PolicyDocument:
+        Statement:
+          - Action:
+              - secretsmanager:GetSecretValue
+            Effect: Allow
+            Resource:
+              Ref: SumoLogicSecret
+          - Action:
+              - kms:Decrypt
+            Effect: Allow
+            Resource:
+              Fn::If:
+                - create_kms_key
+                - Fn::GetAtt:
+                    - SumoLogicKMSKey
+                    - Arn
+                - Ref: AwsKmsKeyARN
+        Version: '2012-10-17'
+      PolicyName: SecretManager-Policy
```

Comment:
Does it not throw error (Policy name exists) if we deploy the same template multiple times in the same region of the account.

Reply:
No, we are good here